libraries/lib-files/FileNames.cpp

```diff
@@ -45,7 +47,14 @@ used throughout Audacity into this one place.
 #include <windows.h>
 #endif
 
+#if defined(__WXGTK__)
+static wxString gOldUnixDataDir;
```

Comment:
I think we can do something like

```
namespace
{
auto GetOldUserDataDir()
{
    static const auto dir = wxFileName::GetHomeDir() + wxT("/.audacity-data");
    return dir;
}

auto OldUserDataDirExists()
{
    static const auto exists = wxDirExists(GetOldUserDataDir());
    return exists;
}
}
```

This will simplify code below

Reply:
Actually, I think I'll just extract all the XDG-specific logic into a separate function and then just inline these 2 functions into it.